esp/services/ws_fs/ws_fsBinding.cpp

```diff
@@ -393,12 +393,15 @@ int CFileSpraySoapBindingEx::downloadFile(IEspContext &context, CHttpRequest* re
         if (!context.validateFeatureAccess(FILE_SPRAY_URL, SecAccess_Full, false))
             throw MakeStringException(ECLWATCH_FILE_SPRAY_ACCESS_DENIED, "Failed to download file. Permission denied.");
 
-        StringBuffer netAddressStr, osStr, pathStr, nameStr;
+        StringBuffer netAddressStr, osStr, pathStr, nameStr, dropZoneName;
         request->getParameter("NetAddress", netAddressStr);
         request->getParameter("OS", osStr);
         request->getParameter("Path", pathStr);
         request->getParameter("Name", nameStr);
+        request->getParameter("DropZoneName", dropZoneName);
 
+        if (getDropZoneScopePermissions(context, dropZoneName, netAddressStr, pathStr) < SecAccess_Read)
+            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Access %s:%s not allowed.", netAddressStr.str(), pathStr.str());
```

Comment:
To help user's debug their permission problems without having to get development involved, can we log more information about the denial?  Such as the LDAP feature name we are checking, or the required access ("Read Access Required") ?

Reply:
I'm not sure SecAccessFlags are meant to be treated as a scalar. Perhaps consider only testing the SecAccess_Read bit, such as 

if (getDropZoneScopePermissions(context, dropZoneName, netAddressStr, pathStr) & SecAccess_Read)

Reply:
> To help user's debug their permission problems without having to get development involved, can we log more information about the denial? Such as the LDAP feature name we are checking, or the required access ("Read Access Required") ?

I will add more information to the error message.

> I'm not sure SecAccessFlags are meant to be treated as a scalar. Perhaps consider only testing the SecAccess_Read bit

We treat the SecAccessFlags as a scalar in many other places: checkWuScopeSecAccess(), SecHandler::authorizeSecReqFeatures(), etc.